NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,125 @@
+
+# ===============================  Fusion rule interface  ==================================
 ⊗(c1::AbstractCategory, c2::AbstractCategory) = fusion_rule(c1, c2)
 
-⊕(c1::AbstractCategory, c2::AbstractCategory) = [c1, c2]
-⊕(cs::Vector{<:AbstractCategory}, c::AbstractCategory) = [cs; c]
-⊕(c::AbstractCategory, cs::Vector{<:AbstractCategory}) = [c; cs]
+function fusion_rule(c1, c2)
+  return fusion_rule(combine_styles(SymmetryStyle(c1), SymmetryStyle(c2)), c1, c2)
+end
+
+function fusion_rule(::SymmetryStyle, c1::C, c2::C) where {C<:AbstractCategory}
+  degen, labels = label_fusion_rule(C, category_label(c1), category_label(c2))
+  return gradedrange(labelled.(degen, C.(labels)))
+end
+
+# abelian case: return Category
+function fusion_rule(::AbelianGroup, c1::C, c2::C) where {C<:AbstractCategory}
+  return C(label_fusion_rule(C, category_label(c1), category_label(c2)))
+end
+
+function fusion_rule(::EmptyCategory, l1::LabelledInteger, l2::LabelledInteger)
+  return labelled(l1 * l2, sector())
+end
+
+function label_fusion_rule(category_type::Type{<:AbstractCategory}, ::Any, ::Any)
+  return error("`label_fusion_rule` not defined for type $(category_type).")
+end
 
-function Base.show(io::IO, cs::Vector{<:AbstractCategory})
-  (length(cs) <= 1) && print(io, "[")
-  symbol = ""
-  for c in cs
-    print(io, symbol, c)
-    symbol = " ⊕ "
-  end
-  (length(cs) <= 1) && print(io, "]")
-  return nothing
+# ================================  GradedAxes interface  ==================================
+# tensor_product interface
+function GradedAxes.fuse_blocklengths(
+  l1::LabelledInteger{<:Integer,<:AbstractCategory},
+  l2::LabelledInteger{<:Integer,<:AbstractCategory},
+)
+  return fuse_blocklengths(combine_styles(SymmetryStyle(l1), SymmetryStyle(l2)), l1, l2)
 end
 
-function trivial(category_type::Type{<:AbstractCategory})
-  return error("`trivial` not defined for type $(category_type).")
+function GradedAxes.fuse_blocklengths(
+  ::SymmetryStyle, l1::LabelledInteger, l2::LabelledInteger
+)
+  fused = label(l1) ⊗ label(l2)
+  v = labelled.(l1 * l2 .* blocklengths(fused), blocklabels(fused))
+  return gradedrange(v)
 end
 
-istrivial(c::AbstractCategory) = (c == trivial(typeof(c)))
+function GradedAxes.fuse_blocklengths(
+  ::AbelianGroup, l1::LabelledInteger, l2::LabelledInteger
+)
+  fused = label(l1) ⊗ label(l2)
+  return labelled(l1 * l2, fused)
+end
 
-function dual(category_type::Type{<:AbstractCategory})
-  return error("`dual` not defined for type $(category_type).")
+function GradedAxes.fuse_blocklengths(
+  ::EmptyCategory, l1::LabelledInteger, l2::LabelledInteger
+)
+  return labelled(l1 * l2, sector())
 end
 
-Base.isless(c1::AbstractCategory, c2::AbstractCategory) = isless(label(c1), label(c2))
+# cast to range
+to_graded_axis(c::AbstractCategory) = to_graded_axis(labelled(1, c))
+to_graded_axis(l::LabelledInteger) = gradedrange([l])
+to_graded_axis(g::AbstractUnitRange) = g
```

Comment:
@ogauthe I think we should change `to_graded_axis` to `to_gradedrange` since it uses the `gradedrange` constructor. Another perspective is that all we know is that it is a graded range being created, it might be in order to use it as an axis of a block array but it could be for some other purpose.